sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -900,24 +883,41 @@ namespace Azure { namespace Storage { namespace DataLake {
           Azure::Core::Context context,
           const CreateOptions& createOptions)
       {
-        auto request = CreateCreateRequest(std::move(url), createOptions);
+        Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Put, url);
+        request.AddHeader(Details::c_HeaderContentLength, "0");
+        request.AddQueryParameter(Details::c_QueryFileSystemResource, "filesystem");
+        if (createOptions.ClientRequestId.HasValue())
+        {
+          request.AddHeader(
+              Details::c_HeaderClientRequestId, createOptions.ClientRequestId.GetValue());
+        }
+        if (createOptions.Timeout.HasValue())
+        {
+          request.AddQueryParameter(
+              Details::c_QueryTimeout, std::to_string(createOptions.Timeout.GetValue()));
+        }
+        request.AddHeader(Details::c_HeaderApiVersionParameter, createOptions.ApiVersionParameter);
+        if (createOptions.Properties.HasValue())
+        {
+          request.AddHeader(Details::c_HeaderProperties, createOptions.Properties.GetValue());
+        }
         return CreateParseResponse(pipeline.Send(context, request));
       }
 
       struct SetPropertiesOptions
       {
-        std::string ClientRequestId; // Provides a client-generated, opaque value with a 1 KB
-                                     // character limit that is recorded in the analytics logs when
-                                     // storage analytics logging is enabled.
-        int32_t Timeout
-            = int32_t(); // The timeout parameter is expressed in seconds. For more
-                         // information, see <a
-                         // href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations">Setting
-                         // Timeouts for Blob Service Operations.</a>
+        Azure::Core::Nullable<std::string>
+            ClientRequestId; // Provides a client-generated, opaque value with a 1 KB character
+                             // limit that is recorded in the analytics logs when storage analytics
+                             // logging is enabled.
+        Azure::Core::Nullable<int32_t>
+            Timeout; // The timeout parameter is expressed in seconds. For more information, see <a
```

Comment:
context

Reply:
This is not exposed to customers as this is protocol layer options.